tests/ImageSharp.Tests/Image/ImageTests.Load_FileSystemPath_PassLocalConfiguration.cs

```diff
@@ -36,7 +36,7 @@ public void Configuration_Path_Agnostic()
                 this.TestFormat.VerifyAgnosticDecodeCall(this.Marker, this.TopLevelConfiguration);
             }
 
-            [Fact]
+            [Fact(Skip = "TODO: Enable when someone tells me how this mocking stuff works.")]
```

Comment:
@tocsoft I really struggled to understand how this worked.

Reply:
Currently the test verifies that the mock `IImageDecoder`s instances `Decode<Rgba32>()` method is called with exactly 2 properties that have reference equality (I believe it reference equality anyway) to `this.TopLevelConfiguration` and `this.DataStream` (which will have been returned from a mock/fake file system configured in `TopLevelConfiguration` when its called with `MockFilePath`).

 For your new code which now wraps all the streams with a `BufferedReadStream` you will need to change the `this.DataStream` parameter to something like `It.Is<Stream>(x=>(BufferedReadStream)x.BaseStream == this.DataStream)` which allows the 'verify' to swap in other sorts of checks. (would require the base stream to be exposed from the buffered read stream)

Reply:
Excellent. That fixed it!